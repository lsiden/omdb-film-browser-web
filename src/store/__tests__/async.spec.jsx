import {
  promiseQueryResults,
  promiseQueryPageFetch,
  promiseFilmDetails
} from 'store/async'
import fetchMock from 'fetch-mock-jest'

// Promise.prototype.catch = fn => new Promise((resolve, reject) => reject(fn))

fetchMock.config.overwriteRoutes = true

afterEach(() => fetchMock.reset())

test('promiseQueryResults()(dispatch) invokes dispatch({type: Actions.REPLACE_FILMS})', () => {
  const dispatch = jest.fn()
  const expectResponse = {
    Search: [
      { imdbID: 1, title: 'Rocky Horror' },
      { imdbID: 2, title: 'Halloween' },
    ],
    totalResults: '4',
  }
  fetchMock.once(/.*/, expectResponse)
  return promiseQueryResults('a query')(dispatch).then(() => {
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})

test('promiseQueryPageFetch()(dispatch)', () => {
  const dispatch = jest.fn()
  const expectResponse = {
    Search: [
      { imdbID: 3, title: 'Rocky Horror II' },
      { imdbID: 4, title: 'Halloween II' },
    ],
    totalResults: '4',
  }
  fetchMock.once(/.*/, expectResponse)
  return promiseQueryPageFetch()(dispatch).then(() => {
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})

test('promiseFilmDetails()(dispatch)', () => {
  const dispatch = jest.fn()
  const expectResponse = {
    filmDetails: {
      title: 'a title',
      director: 'Joe Director',
    },
  }
  fetchMock.once(/.*/, expectResponse)
  return promiseFilmDetails('id')(dispatch).then(() => {
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})
