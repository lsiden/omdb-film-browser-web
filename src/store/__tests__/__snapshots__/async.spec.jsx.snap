// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promiseFilmDetails()(dispatch) 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "isFetching": true,
      },
      "type": "update-is-fetching",
    },
  ],
  Array [
    Object {
      "data": Object {
        "isFetching": false,
      },
      "type": "update-is-fetching",
    },
  ],
  Array [
    Object {
      "data": Object {
        "filmDetails": Object {
          "filmDetails": Object {
            "director": "Joe Director",
            "title": "a title",
          },
        },
      },
      "type": "update-film-details",
    },
  ],
]
`;

exports[`promiseQueryPageFetch()(dispatch) 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "isFetching": true,
      },
      "type": "update-is-fetching",
    },
  ],
  Array [
    Object {
      "data": Object {
        "isFetching": false,
      },
      "type": "update-is-fetching",
    },
  ],
  Array [
    Object {
      "data": Object {
        "films": Array [
          Object {
            "imdbID": 3,
            "title": "Rocky Horror II",
          },
          Object {
            "imdbID": 4,
            "title": "Halloween II",
          },
        ],
        "pageNum": 2,
      },
      "type": "append-films",
    },
  ],
]
`;

exports[`promiseQueryResults()(dispatch) invokes dispatch({type: Actions.REPLACE_FILMS}) 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "isFetching": true,
      },
      "type": "update-is-fetching",
    },
  ],
  Array [
    Object {
      "data": Object {
        "isFetching": false,
      },
      "type": "update-is-fetching",
    },
  ],
  Array [
    Object {
      "data": Object {
        "films": Array [
          Object {
            "imdbID": 1,
            "title": "Rocky Horror",
          },
          Object {
            "imdbID": 2,
            "title": "Halloween",
          },
        ],
        "pageNum": 1,
        "query": "a query",
        "totalResults": "4",
      },
      "type": "update-films",
    },
  ],
]
`;
